components/camera/videosource/webcam.go

```diff
@@ -157,6 +157,11 @@ func (c WebcamConfig) Validate(path string) ([]string, error) {
 			"got illegal negative dimensions for width_px and height_px (%d, %d) fields set for webcam camera",
 			c.Height, c.Width)
 	}
+	if c.FrameRate <= 0 {
+		return nil, fmt.Errorf(
+			"got illegal non-positive dimension for frame rate (%.2f) field set for webcam camera",
+			c.FrameRate)
+	}
```

Comment:
Ah, no, here is has to be explicitly a less than zero check because go populates the struct with a 0 value for floats and integers, this would be a breaking config validation change for all of our users. 

If you had added it to the constructor where I commented, this would not be breaking since it would just populate that frame rate with 0 or whatever mediadevices finds on its own

Reply:
So it is okay if frame rate is set to 0 (and in fact if its not set, it might be set to 0 by mediadevices)? I was trying to align with the "not less than or equal to 0" change with the constructor but I realize now that the config should not invalidate a frameRate = 0 if that could be a default

Reply:
Yep, the logic is not equivalent. One check stops the resource from building at all, another check apparently asks media devices to step in for a parameter that is not there.